arcgis-ios-sdk-samples/Scenes/Animate image overlay/AnimateImagesWithImageOverlayViewController.swift

```diff
@@ -0,0 +1,211 @@
+        }
+    }
+    
+    // MARK: Initialize scene and animation
+    
+    /// Create a scene.
+    ///
+    /// - Returns: A new `AGSScene` object.
+    func makeScene() -> AGSScene {
+        // Create a tiled layer from World Dark Gray Base REST service.
+        let basemapTileURL = URL(string: "https://services.arcgisonline.com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer")!
+        let worldDarkGrayBasemap = AGSArcGISTiledLayer(url: basemapTileURL)
+        // Create an elevation source from Terrain3D REST service.
+        let elevationServiceURL = URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+        let elevationSource = AGSArcGISTiledElevationSource(url: elevationServiceURL)
+        let scene = AGSScene(basemap: AGSBasemap(baseLayer: worldDarkGrayBasemap))
+        // Create a surface and add it to the scene.
+        let surface = AGSSurface()
+        surface.elevationSources = [elevationSource]
+        scene.baseSurface = surface
+        // Create an envelope of the pacific southwest sector for displaying the image frame.
+        let pointForImageFrame = AGSPoint(x: -120.0724273439448, y: 35.131016955536694, spatialReference: .wgs84())
+        pacificSouthwestEnvelope = AGSEnvelope(center: pointForImageFrame, width: 15.09589635986124, height: -14.3770441522488)
+        return scene
+    }
+    
+    func startAnimation(_ fps: Double) {
+        // Invalidate timer to stop previous ongoing animation.
+        frameRateTimer?.invalidate()
+        // Duration or interval between frames in seconds.
+        let duration = 1 / Double(fps)
+        let timer = Timer(timeInterval: duration, repeats: true) { [weak self] _ in
+            self?.setImageFrame()
+        }
+        self.frameRateTimer = timer
+        // Add the timer to common mode run loop, so the timer is not effected by UI events.
+        RunLoop.main.add(timer, forMode: .common)
+    }
+    
+    /// Load the URLs for the overlay images.
+    ///
+    /// - Parameter completion: If it loads successfully, enable the UI components in the completion closure.
+    func loadImageURLs(completion: () -> Void) {
+        // The images need to be added to the project with folder reference.
+        guard let pacificSouthWestURLs = Bundle.main.urls(forResourcesWithExtension: "png", subdirectory: "PacificSouthWest") else {
+            return
+        }
+        // Sort the image URLs by their relative pathnames.
+        imageURLs = pacificSouthWestURLs.sorted { $0.relativeString < $1.relativeString }
+        completion()
+    }
+    
+    /// Set the image at current index to the image overlay.
+    func setImageFrame() {
+        let imageFrame = AGSImageFrame(url: imageURLs[loopCounter.currentIndex], extent: pacificSouthwestEnvelope)
+        imageOverlay.imageFrame = imageFrame
+        loopCounter.next()
+    }
+    
+    func getOpacityString() -> String {
```

Comment:
This function might be of less value, but I'd like to keep format control characters within a single place.